pom.xml

```diff
@@ -268,6 +271,67 @@
         </dependencies>
       </plugin>
 
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>${maven-shade-plugin.version}</version>
+        <configuration></configuration>
+        <executions>
+          <execution>
+            <id>shade-jar-with-dependencies</id>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <phase>package</phase>
+            <configuration>
+              <createDependencyReducedPom>false</createDependencyReducedPom>
+              <useDependencyReducedPomInJar>false</useDependencyReducedPomInJar>
+              <createSourcesJar>true</createSourcesJar>
+              <shadedArtifactAttached>true</shadedArtifactAttached>
+              <shadedClassifierName>shaded</shadedClassifierName>
+              <relocations>
+                <relocation>
+                  <pattern>org.apache</pattern>
+                  <shadedPattern>${shade.prefix}.org.apache.shaded</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.slf4j</pattern>
+                  <shadedPattern>${shade.prefix}.org.slf4j</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.wildfly.common</pattern>
+                  <shadedPattern>${shade.prefix}.org.wildfly.common</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.wildfly.security</pattern>
+                  <shadedPattern>${shade.prefix}.org.wildfly.security</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.jboss</pattern>
+                  <shadedPattern>${shade.prefix}.org.jboss</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.fasterxml.jackson</pattern>
+                  <shadedPattern>${shade.prefix}.com.fasterxml.jackson</shadedPattern>
+                </relocation>
+              </relocations>
+              <transformers>
+                <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+                  <manifestEntries>
+                    <Agent-Class>com.redhat.insights.agent.AgentMain</Agent-Class>
+                  </manifestEntries>
```

Comment:
I noticed any multi-release dependencies we were shading did not get relocated correctly, e.g.:
` META-INF/versions/11/com/fasterxml/jackson/core/io/doubleparser/BigSignificand.class`

This is caused by this bug: https://issues.apache.org/jira/browse/MSHADE-406

I applied some of the workarounds listed there to come up with this solution:

```suggestion
                <!-- TODO Remove the following relocations once this bug is fixed:
                    https://issues.apache.org/jira/browse/MSHADE-406 -->
                <relocation>
                  <pattern>META-INF/versions/9/</pattern>
                  <shadedPattern>META-INF/versions/9/${shade.prefix}/</shadedPattern>
                </relocation>
                <relocation>
                  <pattern>META-INF/versions/11/</pattern>
                  <shadedPattern>META-INF/versions/11/${shade.prefix}/</shadedPattern>
                </relocation>
                <relocation>
                  <pattern>META-INF/versions/17/</pattern>
                  <shadedPattern>META-INF/versions/17/${shade.prefix}/</shadedPattern>
                </relocation>
              </relocations>
              <!-- TODO Revisit the following filters once this bug is fixed:
                    https://issues.apache.org/jira/browse/MSHADE-406 -->
              <filters>
                <filter>
                  <artifact>*:*</artifact>
                  <excludes>
                    <exclude>META-INF/*.SF</exclude>
                    <exclude>META-INF/*.DSA</exclude>
                    <exclude>META-INF/*.RSA</exclude>
                    <exclude>META-INF/MANIFEST.MF</exclude>
                    <exclude>**/module-info.class</exclude>
                    <exclude>META-INF/DEPENDENCIES</exclude>
                  </excludes>
                </filter>
              </filters>
              <transformers>
                <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                  <manifestEntries>
                    <Agent-Class>com.redhat.insights.agent.AgentMain</Agent-Class>
                    <Multi-Release>true</Multi-Release>
                  </manifestEntries>
```